src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -702,161 +779,171 @@ private void AssertCloseWhenNoDocsLeft()
         /// <param name="docsContext"></param>
         /// <param name="sendingCurrentBatchStopwatch"></param>
         /// <returns>Whether succeeded finding any documents to send</returns>
-        private async Task<bool> TrySendingBatchToClient(DocumentsOperationContext docsContext, Stopwatch sendingCurrentBatchStopwatch)
+        private async Task<bool> TrySendingBatchToClient(DocumentsOperationContext docsContext, Stopwatch sendingCurrentBatchStopwatch, SubscriptionBatchStatsScope batchParentScope, int batchStatsIdPerBatch)
         {
             AddToStatusDescription("Starting trying to sent docs to client");
-
             bool anyDocumentsSentInCurrentIteration = false;
-            int docsToFlush = 0;
-            using (var writer = new BlittableJsonTextWriter(docsContext, _buffer))
+            
+            using (batchParentScope.For(SubscriptionOperationScope.DocumentsSent))
```

Comment:
please have a look at ETL if we are using past or present tense.

Reply:
DocumentsSent => SendDocuments